cmds/tui.c

```diff
@@ -2723,13 +2827,17 @@ static void tui_main_loop(struct opts *opts, struct uftrace_data *handle)
 				tui_window_set_middle_next(win, func);
 
 				full_redraw = true;
+				mode = TUI_MODE_REPORT;
 			}
 			break;
 		case 's':
 			if (!tui_window_change(win, &report->win)) {
-				curr_sort_key = (curr_sort_key + 1) %
-						ARRAY_SIZE(report_sort_key);
-				report_setup_sort(report_sort_key[curr_sort_key]);
+				if ((num_selected_report_sort_key = count_selected_report_sort_key()) == 0)
```

Comment:
In general, I don't like an assignment in the condition.  Please assign it outside and compare the variable to 0 in the condition.